PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalBillingCycle.kt

```diff
@@ -0,0 +1,80 @@
+package com.braintreepayments.api.paypal
+
+import android.os.Parcelable
+import com.braintreepayments.api.paypal.PayPalBillingPricing.Companion.toJson
+import kotlinx.parcelize.Parcelize
+import org.json.JSONObject
+
+/**
+ * PayPal recurring billing cycle details.
+ *
+ * @property interval The number of intervals after which a subscriber is charged or billed.
+ * @property intervalCount The number of times this billing cycle gets executed. For example, if
+ * the [intervalCount] is [PayPalBillingInterval.DAY] with an [intervalCount] of 2, the subscription
+ * is billed once every two days. Maximum values [PayPalBillingInterval.DAY] -> 365,
+ * [PayPalBillingInterval.WEEK] -> 52, [PayPalBillingInterval.MONTH] -> 12,
+ * [PayPalBillingInterval.YEAR] -> 1.
+ * @property numberOfExecutions The number of times this billing cycle gets executed. Trial billing
+ * cycles can only be executed a finite number of times (value between 1 and 999). Regular billing
+ * cycles can be executed infinite times (value of 0) or a finite number of times (value between 1
+ * and 999).
+ * @property sequence The sequence of the billing cycle. Used to identify unique billing cycles. For
+ * example, sequence 1 could be a 3 month trial period, and sequence 2 could be a longer term full
+ * rater cycle. Max value 100. All billing cycles should have unique sequence values.
+ * @property startDate The date and time when the billing cycle starts, in Internet date and time
+ * format `YYYY-MM-DDT00:00:00Z`. If not provided the billing cycle starts at the time of checkout.
+ * If provided and the merchant wants the billing cycle to start at the time of checkout, provide
+ * the current time. Otherwise the [startDate] can be in future.
+ * @property isTrial The tenure type of the billing cycle. In case of a plan having trial cycle,
+ * only 2 trial cycles are allowed per plan.
+ * @property pricing The active pricing scheme for this billing cycle. Required if [isTrial] is
+ * false. Optional if [isTrial] is true.
+ */
+@Parcelize
+data class PayPalBillingCycle(
+    var interval: PayPalBillingInterval,
```

Comment:
Do we need to keep these properties as `var`s for java setter support?

Reply:
The optional values need to be `var` for Java setter support, but I updated the required params to `val`. Open to re-considering if we want to continue providing Java setter support in v5 though, but that's likely a separate conversation